src/target/efm32.c

```diff
@@ -572,33 +572,33 @@ bool efm32_probe(target *t)
 		return false;
 	}
 
-	/* Check the EUI is silabs. Identify the Device Identification (DI) version */
-	if (((efm32_v1_read_eui64(t) >> 40) & 0xFFFFFF) == EFM32_V1_DI_EUI_SILABS) {
+	/* Check the OUI in the EUI is silabs or energymicro.
+	 * Use this to identify the Device Identification (DI) version */
+	uint64_t oui24 = ((efm32_v1_read_eui64(t) >> 40) & 0xFFFFFF);
+	if (oui24 == EFM32_V1_DI_EUI_SILABS) {
 		/* Device Identification (DI) version 1 */
 		di_version = 1;
-	} else if (((efm32_v2_read_eui48(t) >> 24) & 0xFFFFFF) == EFM32_V2_DI_EUI_ENERGYMICRO) {
```

Comment:
This change made it impossible for version 2 to ever be detected.

`efm32_v1_read_eui64()` and `efm32_v2_read_eui48()` read from different addresses, so checking `efm32_v1_read_eui64() == EFM32_V2_DI_EUI_ENERGYMICRO` is nonsensical and will never be true.

Unless I'm missing something?